docs/src/lang/option-types.md

```diff
@@ -0,0 +1,386 @@
+example is division over integers, for which `n/0` is undefined.
+
+The basic idea is as follows: given a partial function `f : A -> B`, we form the
+type `Option(B)` by extending `B` with an element representing a missing value,
+`None`, and lift each value `b` in `B` to `Some(b)`, allowing us to represent
+the partial function `pf : A -> Option(B)`, such that, for each `a` in `A`,
+`pf(a) = Some(f(a))` iff `f(a)` is defined, and `None` otherwise.
+
+
+Apalache leverages its support for [variants](./variants.md) to define a
+polymorphic option type along with some common utility functions in the module
+[Option.tla][]. 
+
+The module defines a type alias `$option` as
+
+```tla
+\* @typeAlias: option = Some(a) | None(UNIT);
+```
+
+However, due to the current lack of support for polymorphic aliases, this alias
+has limited utility, and  parametric option types can only be properly expressed
+with the variant type `Some(a) | None(UNIT)`. Nonetheless, in this manual
+page, we will sometimes write `$option(a)` as a shorthand for the type `Some(a)
+| None(UNIT)`.
+
+In the context of TLA+, our encoding of option types is generalized over
+"partial operators", meanings operator which return a value of type
+`$option(a)`. Support for partial functions is supplied by two operators,
+[OptionPartialFun](#OptionPartialFun) and [OptionFunApp](#OptionFunApp).
+
+## Operators
+
+----------------------------------------------------------------------------
+
+### Constructing present optional values
+
+**Notation:** `Some(v)`
+
+**LaTeX notation:** same
+
+**Apalache type:** `(a) => Some(a) | None(UNIT)`, for some type `a`.
+
+**Arguments:** The value `v` of type `a` to be lifted into the
+option.
+
+**Effect:** Produces a new value of the optional type.
+
+**Determinism:** Deterministic.
+
+**Errors:** No errors.
+
+**Example in TLA+:**
+
+```tla
+\* @type: Some(Int) | None(UNIT);
+SomeInt == Some(42)
+```
+
+### Constructing absent optional values
+
+**Notation:** `None`
+
+**LaTeX notation:** same
+
+**Apalache type:** `Some(a) | None(UNIT)`, for some type `a`.
+
+**Arguments:** None
+
+**Effect:** Produces a representation of an absent value.
+
+**Determinism:** Deterministic.
+
+**Errors:** No errors.
+
+**Example in TLA+:**
+
+```tla
+\* @type: Some(Int) | None(UNIT);
+NoInt == None
+```
+----------------------------------------------------------------------------
+
+### Checking for presence or absence of a value
+
+**Notation:** `IsSome(o)` or `IsNone(o)`
+
+**LaTeX notation:** same
+
+**Apalache type:** `(Some(a) | None(UNIT)) => Bool`, for some type `a`.
+
+**Arguments:** One argument: a value of type `$option(a)` for some type `a`.
+
+**Effect:** These operators are `TRUE` or `FALSE` depending on whether the
+optional value is present or absent, in the expected way.
+
+**Determinism:** Deterministic.
+
+**Errors:** No errors.
+
+**Example in TLA+:**
+
+```tla
+TRUE = IsSome(Some(Int)) /\ IsNone(None)
```

Comment:
I think the `Int` set here could be confused with a type; should we put a scalar instead?

```suggestion
TRUE = IsSome(Some(5)) /\ IsNone(None)
```